### ISSUE TITLE
Consider allowing GluegunCommand to be a generic type on Context

### ISSUE THREAD

**Issue by orta** [infinitered/gluegun#338]
Everyone is gonna have different contexts in their apps, so tying a command to just `RunContext` doesn't work

e.g.

```ts
import { GluegunRunContext } from "gluegun/dist/types/domain/run-context"
import { Settings } from "./util/settings"
import { GHCreds } from "./util/promised_ghauth"
import { GluegunCommand } from "gluegun/dist/types/domain/command"

export interface Command extends GluegunCommand {
  run: (context: Context) => void
}

export interface Context extends GluegunRunContext {
  settings: Settings
  githubAuth: GHCreds
}
```

Would raise an error:
```
[ts]
Interface 'Command' incorrectly extends interface 'GluegunCommand'.
  Types of property 'run' are incompatible.
    Type '(context: Context) => void' is not assignable to type '(context: RunContext) => void'.
      Types of parameters 'context' and 'context' are incompatible.
        Type 'RunContext' is not assignable to type 'Context'.
          Property 'settings' is missing in type 'RunContext'.
```

Which makes sense. 

So instead, I could write:

```ts
type Command = GluegunCommand<Settings>
```

I think. Then throughout my app, I can just refer to `Command`.

**Comment by jamonholmgren** [comment]
Yes, this is way better. Thanks!

**Comment by jamonholmgren** [comment]
Closed by #339 .
